package modelo.javabean;

public class Camion extends Vehiculo {
	
	//Atributos
	private double capacidadCargaKg;
	private double cargaOcupadaKg;
	private int numeroEjes;
	
	
	//Métodos heredados
	
	@Override
	public double cargaDisponible() {
		return capacidadCargaKg - cargaOcupadaKg;
	}
	
	@Override
	public double getPorcentajeCarga() {
		return cargaOcupadaKg / capacidadCargaKg * 100;
	}
	
	@Override
	public boolean isLleno() {
		if (cargaOcupadaKg < capacidadCargaKg) {
			return false;
		}
		else {
			return true;
		}
		
	}
	
	
	//Métodos auxiliares
	
	public Camion() {
		super();
	}
	
	public Camion(String matricula, String marca, String modelo, double kilometrosTotales, double consumoLitros100km,
			double capacidadCargaKg, double cargaOcupadaKg, int numeroEjes) {
		super(matricula, marca, modelo, kilometrosTotales, consumoLitros100km);
		this.capacidadCargaKg = capacidadCargaKg;
		this.cargaOcupadaKg = cargaOcupadaKg;
		this.numeroEjes = numeroEjes;
	}
	
	public double getCapacidadCargaKg() {
		return capacidadCargaKg;
	}
	
	public void setCapacidadCargaKg(double capacidadCargaKg) {
		this.capacidadCargaKg = capacidadCargaKg;
	}
	
	public double getCargaOcupadaKg() {
		return cargaOcupadaKg;
	}
	
	public void setCargaOcupadaKg(double cargaOcupadaKg) {
		this.cargaOcupadaKg = cargaOcupadaKg;
	}
	
	public int getNumeroEjes() {
		return numeroEjes;
	}
	
	public void setNumeroEjes(int numeroEjes) {
		this.numeroEjes = numeroEjes;
	}
	
	@Override
	public String toString() {
		return "Camion [matricula=" + matricula + ", marca=" + marca + ", modelo=" + modelo + ", kilometrosTotales="
				+ kilometrosTotales + ", consumoLitros100km=" + consumoLitros100km + ", capacidadCargaKg="
				+ capacidadCargaKg + ", cargaOcupadaKg=" + cargaOcupadaKg + ", numeroEjes=" + numeroEjes + "]";
	}
	
	
	
	
	
	
}	
	